event-adapter/cmd/provider/provider.go

```diff
@@ -14,363 +17,54 @@ limitations under the License.
 package provider
 
 import (
-	"time"
 	"fmt"
-	"strings"
-
+	"github.com/GoogleCloudPlatform/k8s-stackdriver/event-adapter/pkg/config"
+	"github.com/GoogleCloudPlatform/k8s-stackdriver/event-adapter/pkg/provider"
+	"github.com/GoogleCloudPlatform/k8s-stackdriver/event-adapter/pkg/types"
 	"github.com/golang/glog"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/client-go/rest"
-	"k8s.io/apimachinery/pkg/labels"
-	"k8s.io/apimachinery/pkg/runtime/schema"
-	apimeta "k8s.io/apimachinery/pkg/api/meta"
-	"k8s.io/apimachinery/pkg/runtime"
-	"k8s.io/metrics/pkg/apis/custom_metrics"
-	"k8s.io/client-go/pkg/api"
-	_ "k8s.io/client-go/pkg/api/install"
-	"k8s.io/apimachinery/pkg/api/resource"
-	"k8s.io/client-go/kubernetes/scheme"
 	stackdriver "google.golang.org/api/monitoring/v3"
-
-	"k8s.io/k8s-stackdriver-adapter/pkg/provider"
-	"k8s.io/k8s-stackdriver-adapter/pkg/config"
+	"k8s.io/client-go/rest"
+	"time"
 )
 
 type sdService stackdriver.Service
 
+// StackdriverProvider provide all the info for Stackdriver
 type StackdriverProvider struct {
 	restClient rest.Interface
```

Comment:
Are you going to use these fields? Please remove the ones you won't need. For example, values is the map specific to the fake implementation of custom metrics API. Also, please remove the commented-out code.